### ISSUE TITLE
Search box does not recognize `esc` key

### ISSUE THREAD

**Issue by TDA** [bwinton/VerticalTabs#437]
It would be nice to be able to exit the search box on `tabcenter` when I type the `esc` key, right now it does nothing.

**Comment by ericawright** [comment]
I *should* be detecting the `esc` key. Can you let me know what system you're on?
It is meant to detect both when the search box is focused and otherwise, neither of these cases work for you?

**Comment by TDA** [comment]
@ericawright Mac OS El Capitan
MacBook Pro (Retina, 15-inch, Mid 2014)
Firefox 48.0b9

To be clear, I would like to `exit` the search box when pressing escape key, right now it does not seem to exit it :(

**Comment by bwinton** [comment]
@TDA: I think I agree, but where should the focus go when you hit escape?

**Comment by TDA** [comment]
@bwinton So the way I see it:
1. I search for a tab, I do not find it
2. Now I want to open that tab (otherwise I wouldnt be searching for it)
3. So I hit escape on the search box
4. Firefox beautifully transfers focus to the address bar, where I can now open the tab :)

I am not sure how easy/tough it is to do this, but thats a workflow example that I think will be useful to people (especially me) 💃

**Comment by konfou** [comment]
@TDA Doesn't it delete whatever you have written at it that is it cancels current search? Firefox address bar does the same more or less. Nevertheless already opening a new tab (ctrl+n) cancels search, opens new tab and puts focus on the address bar.

**Comment by TDA** [comment]
@konfou I agree that the address bar does the same thing, but the difference is that when I press `esc` while typing in the address bar, I expect it to stay there (possibly cause I mistyped something or changed my mind). The `search tabs` feature is something I use to see if I *already* have something opened and if not, try opening it in the address bar. 
Now if the `search` allowed me to open a new tab with what I am searching for, that would be brilliant, but it doesn't, so the least I could hope for when pressing `esc` is to shift focus to the address bar in a new tab, and be able to open something in that tab. (If thats not possible, I at least expect it to shift focus away from itself, so that I *know* I can use regular browser shortcuts such as `ctrl+n`).
The reason I say this is because, regular shortcuts like `cmd+w` do *not* close the current page while the search bar is focused (and understandable), but `cmd+s` saves the current page. I find this very inconsistent, as this is *not* what the address bar does (both work with the address bar).

The other issue (and I believe this can be a separate issue too) is that, when I manually change focus to the address bar, the search bar is *not* cleared, leading me to reload stuff in the same page, as opposed to in a new tab.
Screen:
<img width="1439" alt="screenshot 2016-07-23 06 57 04" src="https://cloud.githubusercontent.com/assets/432707/17077600/28527ec6-50a3-11e6-85a3-d634f9d92fc6.png">


P.S: `Ctrl+n/t` is *exactly* what I expect when I press `esc` key :) Thank you so much for bringing that up :)

**Comment by fvsch** [comment]
Conventions for what to do on ESC key in search/filter boxes tend to vary. For instance, in Firefox:

- Firefox URL bar: ESC discards a page/result list (if any), and otherwise selects the URL (allowing users to delete it by typing a new value).
- Firefox Search bar: ESC discards a result list (if any), and otherwise does nothing.
- In other programs, including for `<input type=search>` in most browsers (everyone except Fx, if my memory serves), ESC empties the search box.

Moving the focus to a different field is an opinionated behavior that is not consistent with common practices, so I’d advise against it.

To answer @TDA’s use case:

> 1. I search for a tab, I do not find it
> 2. Now I want to open that page in a tab (otherwise I wouldnt be searching for it)
> 3. So I hit escape on the search box
> 4. Firefox beautifully transfers focus to the address bar, where I can now open the tab :)

You can:
- Hit Cmd+T (or Ctrl+T) to open a new tab. The focus will be in the URL bar, and Tab Center will reset its search/filtering.
- Or hit Cmd+L (or Ctrl+L) to focus the URL bar for the currently active tab/page. In that case Tab Center will not reset its search/filtering, and maybe it should, but I think that’s a scenario that won’t happen often since users are more likely to go for the Cmd+T option.

So you just need to replace your "3. I hit ESC" step by "3. I hit Cmd+T".

> The other issue (and I believe this can be a separate issue too) is that, when I manually change focus to the address bar, the search bar is not cleared, leading me to reload stuff in the same page, as opposed to in a new tab (and not allowing me to see my tabs).

For the "leading me to reload stuff in the same page", that's just what browsers do, so it's not related to Tab Center. For clearing the Tab Center filtering when you focus the URL bar, that’s a valid use case but on the other hand other users may want to keep the filtering and use the page(s). Magically discarding the filtering might hurt those other use cases.

Note that there is a "N more tabs…" entry when the tab list is filtered, and clicking it clears the filtering.

Bottom line: I think the current behavior is alright.

**Comment by grahamperrin** [comment]
This seems to be fixed. 

Tested: 1.34.0 on Firefox 53.0 (64-bit) on TrueOS Desktop.
